buildSrc/src/test/groovy/org/gradle/binarycompatibility/AcceptedApiChangesJsonFileManagerTest.groovy

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.binarycompatibility
+
+import org.junit.Rule
+import org.junit.rules.TemporaryFolder
+import spock.lang.Specification
+import spock.lang.Subject
+
+class AcceptedApiChangesJsonFileManagerTest extends Specification {
+
+    @Rule
+    TemporaryFolder temporaryFolder = new TemporaryFolder()
+
+    @Subject AcceptedApiChangesJsonFileManager jsonFileManager = new AcceptedApiChangesJsonFileManager()
+
+    def jsonFile
+
+    def setup() {
+        jsonFile = temporaryFolder.newFile('acceptedChanges.json')
+    }
+
+    def "can clean existing API changes"() {
+        given:
+        jsonFile << existingAcceptedApiChanges()
+
+        when:
+        jsonFileManager.emptyAcceptedApiChanges(jsonFile)
+
+        then:
+        jsonFile.text == nonExistentAcceptedApiChanges()
+    }
+
+    def "can clean non-existing API changes"() {
+        given:
+        jsonFile << nonExistentAcceptedApiChanges()
+
+        when:
+        jsonFileManager.emptyAcceptedApiChanges(jsonFile)
+
+        then:
+        jsonFile.text == nonExistentAcceptedApiChanges()
+    }
+
+    static String existingAcceptedApiChanges() {
+        """
+            {
+                "baseVersion": "4.1-20170714213039+0000",
```

Comment:
`baseVersion` isn't part of the file any more. How did you come up with the example?

Reply:
Got this from the linked commit in the release document under ["Branching for the Release"](https://docs.google.com/document/d/1h_Me4H4PfCRNUg5KLHht9zdSMNmMeOmQycvSyeBtyJ8/edit#heading=h.3znysh7) (point 4): https://github.com/gradle/gradle/blob/d4133a7e85046ed2c6f677739043487f38a83898/subprojects/distributions/src/changes/accepted-public-api-changes.json#L4-L39. I will remove the `baseVersion` if that doesn't exist anymore.